sdk/java/pulumi/src/main/java/com/pulumi/core/internal/Urn.java

```diff
@@ -99,7 +105,7 @@ public static Urn parse(String urn) {
                 () -> format("expected urn to start with '%s', got: '%s'", Prefix, urn)
         );
 
-        var urnParts = urn.split(PartsSeparator);
+        var urnParts = urn.split(PartsSeparatorRegex, -1); // -1 avoids dropping trailing empty strings
```

Comment:
wow, thanks for spotting that, that's surprising! not a good thing from an API ;)